### PR TITLE
helm: Make prometheus job name compatible with grafana dashboard (#907)

### PR DESCRIPTION
**Pull request by zerda**
I have fixed a glitch in kube-prometheus and synced with helm in separate commits. Most of changes are come from etcd dashboard.

All jobs, rules and dashboards are working fine except the etcd one. It seems not available in on-premise cluster built with kubeadm.

### CODE REVIEW

**Comment by brancz** [comment]
Yeah, with etcd, one must make sure themselves, that etcd metrics are collected, there is unfortunately no one size fits all solution today.

**Comment by brancz** [comment]
cc @gianrubio

File: contrib/kube-prometheus/manifests/grafana/grafana-dashboards.yaml [hunk 1]
```diff
@@ -7495,4 +7495,3 @@ data:
         "type": "prometheus",
         "url": "http://prometheus-k8s.monitoring.svc:9090"
     }
----
```
Comment (zerda) [inline contrib/kube-prometheus/manifests/grafana/grafana-dashboards.yaml]: I am aware this file is generated by `contrib/kube-prometheus/hack/grafana-dashboards-configmap-generator/bin/grafana_dashboards_generate.sh`, manually editing is not recommended. I will fix that asap.

**Comment by zerda** [comment]
I fixed the sync script, and reverted the grafana-dashboards.yaml modification. Sorry for any inconvenience caused.

**Comment by gianrubio** [comment]
lgtm, thanks @zerda

**Pull request merged**